jobs/rabbitmq-server/templates/prom_scraper_config.yml.erb

```diff
@@ -1,6 +1,19 @@
 ---
 port: <%= if p("rabbitmq-server.management_tls.enabled") then 15691 else 15692 end %>
-source_id: <%= if p('rabbitmq-server.cluster_name') == "" then 'rabbit@localhost' else p('rabbitmq-server.cluster_name') end %>
+
+<%
+if p('rabbitmq-server.prom_scraper_source_id') != "" then
+  source_id = p('rabbitmq-server.prom_scraper_source_id')
+  if p('rabbitmq-server.ensure_log_cache_compatibility') and p('rabbitmq-server.prom_scraper_source_id').length > 48 then
+    raise 'prom_scraper source_id must be 48 characters or less'
+  end
+elsif p('rabbitmq-server.cluster_name') == "" then
+  source_id = 'rabbit@localhost'
+else
+ source_id = p('rabbitmq-server.cluster_name')
+end
+-%>
+source_id: <%= source_id %>
 instance_id: <%= if p('rabbitmq-server.create_swap_delete') == true then "'rabbit@#{spec.address}'" else "'rabbit@#{Digest::MD5.hexdigest(spec.ip)}'" end %>
```

Comment:
Don't know if you plan to do it in a follow-on PR or not, but this is missing the instance_id change. It's probably good to lump both together in the same PR so both changes end up in the same BOSH release.